api/wallet.go

```diff
@@ -106,35 +130,89 @@ func (srv *Server) walletHandler(w http.ResponseWriter, req *http.Request) {
 	unlockConditions, err := srv.wallet.NextAddress()
 	if err != nil {
 		writeError(w, "error after call to /wallet/seed: "+err.Error(), http.StatusBadRequest)
 		return
 	}
-	writeJSON(w, WalletAddressPOST{
+	writeJSON(w, WalletAddressGET{
 		Address: unlockConditions.UnlockHash(),
 	})
 }
 
 // walletAddressHandler handles API calls to /wallet/address.
 func (srv *Server) walletAddressHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method == "" || req.Method == "GET" {
+		srv.walletAddressHandlerGET(w, req)
+		return
+	}
+	writeError(w, "unrecognized method when calling /wallet/address", http.StatusBadRequest)
+}
+
+// walletBackupHandlerPOST handles a POST call to /wallet/backup
+func (srv *Server) walletBackupHandlerPOST(w http.ResponseWriter, req *http.Request) {
+	err := srv.wallet.CreateBackup(req.FormValue("Filepath"))
+	if err != nil {
+		writeError(w, "error after call to /wallet/backup: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+	writeSuccess(w)
+}
+
+// walletBackupHandler handles API calls to /wallet/backup.
+func (srv *Server) walletBackupHandler(w http.ResponseWriter, req *http.Request) {
 	if req.Method == "POST" {
-		srv.walletAddressHandlerPOST(w, req)
-	} else {
-		writeError(w, "unrecognized method when calling /wallet/address", http.StatusBadRequest)
+		srv.walletBackupHandlerPOST(w, req)
+		return
 	}
+	writeError(w, "unrecognized method when calling /wallet/backup", http.StatusBadRequest)
+}
+
+// walletEncryptHandlerPOST handles a POST call to /wallet/encrypt.
+func (srv *Server) walletEncryptHandlerPOST(w http.ResponseWriter, req *http.Request) {
+	var encryptionKey crypto.TwofishKey
+	if req.FormValue("EncryptionPassword") != "" {
+		encryptionKey = crypto.TwofishKey(crypto.HashObject(req.FormValue("EncryptionPassword")))
+	}
+	seed, err := srv.wallet.Encrypt(encryptionKey)
+	if err != nil {
+		writeError(w, "error when calling /wallet/encrypt: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	dictID := mnemonics.DictionaryID(req.FormValue("Dictonary"))
+	if dictID == "" {
+		dictID = "english"
+	}
+	seedStr, err := modules.SeedToString(seed, dictID)
+	if err != nil {
+		writeError(w, "error when calling /wallet/encrypt: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+	writeJSON(w, WalletEncryptPOST{
+		PrimarySeed: seedStr,
+	})
+}
+
+// walletEncryptHandler handles API calls to /wallet/encrypt.
+func (srv *Server) walletEncryptHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method == "POST" {
+		srv.walletEncryptHandlerPOST(w, req)
+		return
+	}
+	writeError(w, "unrecognized method when calling /wallet/encrypt", http.StatusBadRequest)
 }
 
 // walletHistoryHandlerGET handles a GET request to /wallet/history.
 func (srv *Server) walletHistoryHandlerGET(w http.ResponseWriter, req *http.Request) {
-	start, err := strconv.Atoi(req.FormValue("startHeight"))
+	start, err := strconv.Atoi(req.FormValue("StartHeight"))
```

Comment:
It looks like a lot of parameters were capitalized. This definitely isn't standard practice. I would go as far as to suggest automatically lower-casing all query string params as they come in.